utils/build_rpms_from_scratch.sh

```diff
@@ -6,52 +6,126 @@ set -uex
 #
 # Description:
 # This script sets up an environment to build and install RPM packages for DAOS
-# and its dependencies on a Rocky Linux system. It automates the installation
-# of required tools, package dependencies, and specific repositories, ensuring
-# that each RPM is built in a temporary location and copied to a destination
-# directory.
-#
-# Usage:
-#   Run the script in a Rocky Linux system
+# and its dependencies. The script serves as an example of what dependencies are
+# needed for the DAOS RPMs, but it has been tested on rockylinux 8 and 9, and on
+# opensuse/leap 15.4, 15.5, and 15.6.
 #
 # Arguments:
 #   1. Destination for built RPMs (default: "$HOME/rpm/")
 #
 ###############################################################################
 
+# Get distro information so can be used throughout the script globally
+. /etc/os-release
+OS_NAME=$NAME
+OS_VERSION=$VERSION_ID
+OS_ID=$ID
+
+# -----------------------------------------------------------------------------
+# This first set of functions serve to abstract away some OS specifics. It is
+# assumed that by the time these functions are called, the OS and Version
+# have been verified
+# -----------------------------------------------------------------------------
+
 # Set up the temporary local repository configuration with GPG check disabled
 setup_local_repo() {
   local rpms_dst=$1
 
+  if [[ "$OS_ID" == "opensuse-leap" ]]; then
+    repo_file=/etc/zypp/repos.d/local-rpms.repo
+  else
+    repo_file=/etc/yum.repos.d/local-rpms.repo
+  fi
+
   echo "[local-rpms]
 name=Local RPM Repository
 baseurl=file://$rpms_dst
 enabled=1
-gpgcheck=0" > /etc/yum.repos.d/local-rpms.repo
+gpgcheck=0" > "$repo_file"
 }
 
 # Clean up the temporary repository configuration
 cleanup_local_repo() {
-  rm -f /etc/yum.repos.d/local-rpms.repo
+  if [[ "$OS_ID" == "opensuse-leap" ]]; then
+    rm -f /etc/zypp/repos.d/local-rpms.repo
+  else
+    rm -f /etc/yum.repos.d/local-rpms.repo
+  fi
 }
 
-dnf_builddep() {
-  local rpms_dst=$1
-  local spec_file="$2"
-  # dnf --nogpgcheck --repofrompath rpms,"$rpms_dst" --refresh builddep -y "$spec_file"
-  dnf --refresh builddep -y "$spec_file"
+install_dependencies() {
+  local spec_file="$1"
+
+  if [[ "$OS_ID" == "opensuse-leap" ]]; then
+   dependencies=$(rpmspec -q --buildrequires "$spec_file" | grep -v "^rpmlib" | grep -v "^/" | tr '\n' ' ')
+      if [ -n "$dependencies" ]; then
+          # Install dependencies using zypper
+          echo "Installing build dependencies: $dependencies"
+          zypper install --allow-vendor-change -y $dependencies
+      else
+          echo "No build dependencies found in the spec file."
+      fi
+  else
+    dnf --refresh builddep -y "$spec_file"
```

Comment:
Only concern here is that `--refresh` is going to refresh all of the OS distros which are going to be expensive, NOOP refreshes.  The alternative is to run:
`dnf --disablerepo=\* --enablerepo=local-rpms makecache` either here or in `l_createrepo()` after the `createrepo`, much like you are doing for `zypper`.